### ISSUE TITLE
None gradients when using tf gradient tape

### ISSUE THREAD

**Issue by Senantq** [tensorflow/tensorflow#75032]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

binary

### TensorFlow version

tf 2.17.0

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 24.04 LTS

### Python version

3.11.9

### Bazel version

binary

### GCC/compiler version

binary

### CUDA/cuDNN version

12.3 (pip install tensorflow[and-cuda)==2.17)

### GPU model and memory

RTX A4500 laptop, 16 gigs

### Current behavior?

I tried to load a trained model using tensorflow and keras in order to compute some saliency maps. To do so, I need to compute the gradients with respect to the inputs. However, it returns None gradients whereas it worked fine before, e.g., with tf 2.10.
A reprex with the model weights, a datafile and a minimal version of the code can be find using the following google drive link:

### Standalone code to reproduce the issue

```shell
https://drive.google.com/file/d/1fcind0XWxdghlKvN237y0L177JbEgOcV/view?usp=sharing
```

**Comment by Venkat6871** [comment]
Hi **@Senantq** ,
I tried to run your code on Colab using TF v2.17, but it is asking for a directory. Could you please provide the path related to your issue?
Thank you!
